packages/ui-pattern-block/src/UIPatternBlock.tsx

```diff
@@ -1,25 +1,239 @@
+import { SandpackLayout, SandpackPreview, SandpackProvider } from '@codesandbox/sandpack-react';
 
 import 'tailwindcss/tailwind.css';
 import '@frontify/guideline-blocks-settings/styles';
 
-import { type Settings, sandpackThemeValues } from './types';
+import { Alignment, Height, type Settings, sandpackThemeValues } from './types';
+import {
+    DEFAULT_BLOCK_SETTINGS,
+    EDITOR_CLASSES,
+    centeringCss,
+    getBackgroundCss,
+    getCssToInject,
+    getDefaultFilesOfTemplate,
+    getHeightStyle,
+    getPaddingStyle,
+    getParsedDependencies,
+    getRadiusValue,
+    getScriptToInject,
+    initialActiveFile,
+} from './helpers';
+import { useDebounce } from './hooks';
+import { Captions, CodeEditor, ExternalDependencies, NPMDependencies, ResponsivePreview } from './components';
 
 export const UIPatternBlock = ({ appBridge }: BlockProps): ReactElement => {
-    const [blockSettings] = useBlockSettings<Settings>(appBridge);
-    const { sandpackTemplate, sandpackTheme } = blockSettings;
+    const [blockSettings, setBlockSettings] = useBlockSettings<Settings>(appBridge);
+    const {
+        sandpackTemplate,
+        sandpackTheme,
+        files,
+        alignment,
+        dependencies: blockDependencies,
+        paddingChoice,
+        paddingCustom,
+        hasCustomPadding,
+        heightChoice,
+        customHeightValue,
+        isCustomHeight,
+        showResetButton,
+        showSandboxLink,
+        showResponsivePreview,
+        showCode,
+        showNpmDependencies,
+        showExternalDependencies,
+        shouldCollapseCodeByDefault,
+        shouldCollapseDependenciesByDefault,
+        backgroundColor,
+        hasBackground,
+        borderColor,
+        borderStyle,
+        borderWidth,
+        hasBorder,
+        hasRadius,
+        radiusChoice,
+        radiusValue,
+        description,
+        title,
+    } = { ...DEFAULT_BLOCK_SETTINGS, ...blockSettings };
+    const { debounce } = useDebounce();
+    const isEditing = useEditorState(appBridge);
+    const [isResponsivePreviewOpen, setIsResponsivePreviewOpen] = useState(false);
+    const [dependencies, setDependencies] = useState(blockDependencies);
+    const [reset, setReset] = useState(false);
+
+    const cssToInject = useMemo(() => {
+        const alignmentCss = alignment === Alignment.Center ? centeringCss : '';
+        const backgroundCss = hasBackground ? getBackgroundCss(backgroundColor) : '';
+        const hasAutoHeight = !isCustomHeight && heightChoice === Height.Auto;
+        return getScriptToInject(getCssToInject(hasAutoHeight, alignmentCss, backgroundCss));
+    }, [hasBackground, backgroundColor, alignment, isCustomHeight, heightChoice]);
+
+    const templateFiles = useMemo(
+        () => ({ ...getDefaultFilesOfTemplate(sandpackTemplate), ...files?.[sandpackTemplate] }),
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [sandpackTemplate, cssToInject, dependencies, reset],
```

Comment:
```suggestion
        [sandpackTemplate, files],
```

What I am missing that its not like that?

Reply:
so every time this variable changes the sandpack provider is remounted. That's how it works internally. If the `files` variable was added to the dependency array, it would remount on every keystroke. 

The other dependencies are needed, because when e.g. dependencies change, then the sandbox is remounted, and we need to get the latest saved files, so we need to rerun the usememo.

I changes the dependency array to be the same array I use for the key attribute, so they always match.